### PR TITLE
Merge policy object

### PR DESCRIPTION
**Pull request by phochard**

### CODE REVIEW

File: src/core/primitives.rs [hunk 1]
```diff
@@ -192,11 +194,13 @@ pub fn setup(
             s2,
             subkeys: sub_sk,
             kmac_key,
+            policy: policy.clone()
         },
         MasterPublicKey {
             g1,
             g2,
             subkeys: sub_pk,
+            policy : policy.clone()
```
[the remark this answers is not in the document]
Comment (phochard) [reply src/core/primitives.rs]: I don't know how to resolve this error : 
```
error[E0382]: use of moved value: `policy`
   --> src/core/primitives.rs:202:13
    |
177 |     let policy = Policy::new();
    |         ------ move occurs because `policy` has type `PolicyV2`, which does not implement the `Copy` trait
...
196 |             policy,
    |             ------ value moved here
...
202 |             policy,
    |             ^^^^^^ value used here after move
    |
help: consider cloning the value if the performance cost is acceptable
    |
196 |             policy: policy.clone(),
    |                   ++++++++++++++++
```
Could you give me a hint please ?

File: examples/encrypt.rs [hunk 2]
```diff
@@ -30,10 +30,10 @@ fn generate_new(
         use cosmian_crypto_core::bytes_ser_de::Serializable;
         let config: GeneralPurposeConfig = GeneralPurposeConfig::default();
         let transcoder: GeneralPurpose = GeneralPurpose::new(&STANDARD, config);
-        println!(
+        println!(
```
Comment (tbrezot) [inline examples/encrypt.rs:33]: You need to setup your editor not to leave white spaces behind.

File: src/abe_policy/attribute.rs [hunk 3]
```diff
@@ -146,7 +146,7 @@ impl TryFrom<&str> for Attribute {
 
 impl std::fmt::Display for Attribute {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        write!(f, "{}::{}", self.dimension, self.name)
+        write!(f, "{}:  :{}", self.dimension, self.name)
```
Comment (tbrezot) [inline src/abe_policy/attribute.rs:149]: No, `::` is on purpose
```suggestion
        write!(f, "{}::{}", self.dimension, self.name)
```

File: src/core/primitives.rs [hunk 4]
```diff
@@ -97,10 +100,8 @@ pub fn usk_keygen(
     let coordinate_keys = msk
         .get_latest_coordinate_sk(coordinates.into_iter())
         .collect::<Result<RevisionVec<_, _>, Error>>()?;
-
```
Comment (tbrezot) [inline src/core/primitives.rs:100]: Why did you remove these empty lines?

File: src/core/serialization/mod.rs [hunk 5]
```diff
@@ -526,6 +536,43 @@ impl Serializable for CleartextHeader {
...
+            //         .iter()
+            //         .map(|(s, dim)| {
+            //             to_leb128_len(s.len())
+            //                 + s.len()
+            //                 + dim
+            //                     .attributes()
+            //                     .map(|d| to_leb128_len(d.id as usize) + 2)
+            //                     .sum::<usize>()
+            //         })
+            //         .sum::<usize>()
```
Comment (tbrezot) [inline src/core/serialization/mod.rs:559]: All commented code is to be removed. you can keep it somewhere in a branch but not in MR.
```suggestion
```

File: src/test_utils/non_regression.rs [hunk 6]
```diff
@@ -178,7 +172,7 @@ impl NonRegressionTestVector {
         let reg_vectors = Self {
             public_key: transcoder.encode(mpk.serialize()?),
             master_secret_key: transcoder.encode(msk.serialize()?),
-            policy: transcoder.encode(<Vec<u8>>::try_from(&policy).unwrap()),
+            policy: policy.to_string(),
```
Comment (tbrezot) [inline src/test_utils/non_regression.rs:175]: Remove the policy field instead (no need to serialize it in the regression vector since it is already present in the serialized MPK):
```suggestion
```

File: src/test_utils/statics.rs
Comment (tbrezot) [inline src/test_utils/statics.rs:1]: empty file to be removed

File: src/core/primitives.rs (hunk 4 above)
Comment (phochard) [reply src/core/primitives.rs:100]: I have no idea, sorry 😓

**Review by tbrezot — APPROVED** [review]
